File: erezervisi_api/eRezervisi/eRezervisi.Common.Shared/Pagination/BasePagedRequest.cs
using System.Text.Json.Serialization;

namespace eRezervisi.Common.Shared.Pagination
{
    public class BasePagedRequest
    {
        public long Page { get; set; }
        public long PageSize { get; set; }
        public string SearchTerm { get; set; } = string.Empty;
        public string OrderByColumn { get; set; } = string.Empty;
        public string OrderBy { get; set; } = "desc";
    }

    public class BaseGetAllRequest
    {
        public string SearchTerm { get; set; } = string.Empty;
        [JsonIgnore]
        public string SearchTermLower => SearchTerm.ToLower();
    }
}

File: erezervisi_mobile/erezervisi_mobile/lib/models/requests/base/base_paged_request.dart
class BasePagedRequest {
  late num page;
  late num pageSize;
  late String searchTerm;
  late String orderByColumn;
  late String orderBy;

  BasePagedRequest(
      {required this.page,
      required this.pageSize,
      required this.searchTerm,
      required this.orderByColumn,
      required this.orderBy});

  Map<String, dynamic> toJson() {
    return {
      'page': page,
      'pageSize': pageSize,
      'searchTerm': searchTerm,
      'orderByColumn': orderByColumn,
      'orderBy': orderBy
    };
  }
}
